packages/api/src/utils/github/buildProjectsQuery.ts

```diff
@@ -0,0 +1,56 @@
+import { Project } from '../../entities/Project';
+import { env } from '../../env';
+import logger from '../../logger';
+
+interface Repository {
+  id: string;
+  nameWithOwner: string;
+}
+
+interface RepositoryResponse {
+  data: {
+    nodes: Repository[];
+  };
+}
+
+export const buildProjectsQuery = async (projects: Project[]) => {
+  const repoNodeIds: String[] = [];
+  projects.map((project) => repoNodeIds.push(project.nodeID));
+
+  const findReposQuery = JSON.stringify({
+    query: `
+      query FindRepositories($repoIds: [ID!]!) { 
+        nodes(ids: $repoIds){
+          ... on Repository {
+            id
+            nameWithOwner
+          }    
+        }
+      }
+    `,
+    variables: {
+      repoIds: repoNodeIds,
+    },
+  });
+
+  try {
+    const fetchRes = await fetch('https://api.github.com/graphql', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `bearer ${env.githubToken}`,
+      },
+      body: findReposQuery,
+    });
+
+    const { data: responseData }: RepositoryResponse = await fetchRes.json();
+
+    const projectsQuery = responseData.nodes.map((repo) => `repo:${repo.nameWithOwner}`).join(' ');
+
+    return projectsQuery;
+  } catch (error) {
+    const errorMessage = 'There was an issue getting repository info from graphql';
+    logger.error(errorMessage, error);
+    return '';
```

Comment:
Since the happy path here is a string, I'm a little concerned with returning one if things break... I think we need to throw an error here or return `null` and then handle it wherever this is being used